langchain/src/vectorstores/weaviate.ts

```diff
@@ -164,12 +164,22 @@ export class WeaviateStore extends VectorStore {
     });
 
     try {
-      await this.client.batch
+      const responses = await this.client.batch
```

Comment:
So this was buggy before and `client.batch` doesn't throw at all?

Reply:
Ahm, no. As written in the PR description, certain errors were already catched correctly. Just try and address a server, which does not exist, e.g. `http://localhost:8080` -> that results in an error `FetchError: request to http://localhost:8080/v1/batch/objects failed, reason: connect ECONNREFUSED ::1:8080` also without the change of this PR.

However, if the server processes the request and sends a response, it still may contain an error. In my cases these were always 4xx errors, mostly 422 - Unprocessable Entity. However, in principle I could also imagine that Weaviate could respond with 5xx errors this way.
These kind of errors are catched with the changes of this PR.